external/fv3fit/fv3fit/data/batches.py

```diff
@@ -0,0 +1,53 @@
+import dataclasses
+import os
+from typing import Optional, Sequence
+import loaders
+import loaders.typing
+from loaders.batches.save import save_batches
+from fv3fit.tfdataset import tfdataset_from_batches
+import tensorflow as tf
+import logging
+from .base import TFDatasetLoader
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class FromBatches(TFDatasetLoader):
+
+    batches_loader: loaders.BatchesLoader
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "FromBatches":
+        return FromBatches(loaders.BatchesLoader.from_dict(d))
+
+    def get_data(
+        self, local_download_path: Optional[str], variable_names: Sequence[str],
+    ) -> tf.data.Dataset:
+        """
+        Args:
+            local_download_path: if provided, cache data locally at this path
+            variable_names: names of variables to include when loading data
+        Returns:
+            dataset containing requested variables
```

Comment:
Updated. There are relatively few requirements about shapes and dtype - no restriction on dtype, and the only restriction on shape is that the data should be "batched" (to follow what is done currently by the from-batches loading, since it's trivial to call `.unbatch` in a training function that calls for it, but not trivial to batch into the sizes set by the batches loader config).

Reply:
I updated it on the base classes but not the subclasses, that's fixed now.